src/razor/src/csharp/csharpProjectedDocument.ts

```diff
@@ -150,4 +168,18 @@ export class CSharpProjectedDocument implements IProjectedDocument {
     private setContent(content: string) {
         this.content = content;
     }
+
+    private updateContent(edits: ServerTextChange[]) {
+        if (edits.length === 0) {
+            return;
+        }
+
+        let content = this.content;
+        for (const edit of edits.reverse()) {
```

Comment:
I'm surprised this works, given elsewhere the edits are concated together. Wouldn't we need to insert subsequent edits at position 0?

Though that could be at odds with the Roslyn algorithm that processes the same edits?

Reply:
that's a good point. I don't think I've actually hit a scenario where they got concat together in testing so probably missed this. Idk why reverse was used initially. Let me look...

Reply:
Reverse makes sense for a single batch of ordered changes, it means you don't need to track forward when applying edits earlier in the file, and adjust positions later in the file. But the batches would need to be applied in order.

Reply:
I think I can just update to applying stored edits first instead of doing a concat

Reply:
This should be better now. I updated to keep track of `Updates` as individual units sent from Razor. How they get applied  remains consistent and the edits being returned on ordered to be sent back to Roslyn